src/core/validate-graph.ts

```diff
@@ -60,6 +60,19 @@ export function validatePackageURL(pkg: types.PkgInfo): void {
         );
         break;
 
+      // The PURL spec for Linux distros does not include the source in the name.
+      // This is why we relax the assertion here and match only on the package name:
+      // <source name>/<package name> - we omit the source name
+      // For now, make this exception only for deb to cover a support case.
+      case 'deb': {
+        const pkgName = pkg.name.split('/').pop();
```

Comment:
Sorry that I'm just coming up with this - I sincerely hope this will be the last comment :) 

Would it make sense to check for the source package if the upstream is defined?
E.g. ensuring that `pkgSource == purlPkg.qualifiers['upstream'].split('@').head()`?

Or is that too much?

Reply:
Updated!